python/cudf/cudf/pandas/_wrappers/pandas.py

```diff
@@ -260,18 +260,14 @@ def Index__new__(cls, *args, **kwargs):
     return self
 
 
-def name(self):
-    return self._fsproxy_wrapped._name
-
-
 def Index__setattr__(self, name, value):
```

Comment:
With the usage of `_FastSlowAttribute` I don't think we need `Index__setattr__` anymore.

Reply:
I agree, except for MultiIndex

Reply:
Wouldn't defining `names` Fastslow attribute for MultiIndex just do the job?

Reply:
`names` attr in Pandas is set with [`property` like this](https://github.com/pandas-dev/pandas/blob/d9cdd2ee5a58015ef6f4d15c7226110c9aab8140/pandas/core/indexes/multi.py#L1632). I think`_set_names` is not being picked up for some reason

Reply:
pd.MultiIndex([]).names = ... isn't doing anything

Reply:
No I think we still need `Index__setattr__` in Index types. This issue address the problem [described here](https://github.com/rapidsai/cudf/issues/14524#issuecomment-1833745159).

Reply:
> No I think we still need `Index__setattr__` in Index types. This issue address the problem [described here](https://github.com/rapidsai/cudf/issues/14524#issuecomment-1833745159).

I feel if we do this for one attribute we will have to end up doing the same for every attribute, which is why I think there could be a better approach to solve this problem. Let's pair on this tomorrow ?